compaction/compaction_manager.cc

```diff
@@ -1504,7 +1504,7 @@ future<> compaction_manager::perform_cleanup(owned_ranges_ptr sorted_owned_range
             const auto candidates = get_candidates(t);
             std::copy_if(candidates.begin(), candidates.end(), std::back_inserter(sstables), [&sorted_owned_ranges, schema] (const sstables::shared_sstable& sst) {
                 seastar::thread::maybe_yield();
-                return sst->needs_cleanup(*sorted_owned_ranges, schema);
+                return sst->mark_for_cleanup(*sorted_owned_ranges, schema);
```

Comment:
I'm a uneasy about this (and the patch that makes needs_cleanup a member). Instead of sstable being a building block, suddenly it knows about owned ranges and cleanup.

If the function was called intersects(ranges), it remains neutral in terms of knowing about topology, but knowing about cleanup is a little too much. Can we mark it at a higher level (compaction_manager) instead?

Reply:
btw, it's odd that we need the schema, and that we need to pass the schema. The sstable knows its own schema!

Reply:
yes, in the next version it will be in compaction_manager.

Reply:
> btw, it's odd that we need the schema, and that we need to pass the schema. The sstable knows its own schema!

True.  We don't need it.
Will be removed in next version.

Reply:
moved to compaction_manager as update_sstable_cleanup_state